File: algorithm/src/main/java/baekjoon/p05073/Main.java
package baekjoon.p05073;

import java.util.Arrays;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            int a = sc.nextInt();
            int b = sc.nextInt();
            int c = sc.nextInt();
            if (a == 0 && b == 0 && c == 0) {
                break;
            }

            System.out.println(solve(a, b, c));
        }
    }

    private static String solve(int a, int b, int c) {
        int[] arr = {a, b, c};
        Arrays.sort(arr);

        if (arr[0] + arr[1] > arr[2]) {
            if (arr[0] == arr[2]) {
                return "Equilateral";
            } else if (arr[0] == arr[1] || arr[1] == arr[2]) {
                return "Isosceles";
            } else {
                return "Scalene";
            }
        } else {
            return "Invalid";
        }
    }

}

File: algorithm/src/main/java/baekjoon/p05073/Main.kt
package baekjoon.p05073

fun main(args: Array<String>) {
    while (true) {
        val inputs = readLine()!!.split(" ").map { it.toInt() }
        if (inputs.all { it == 0 }) {
            break
        }
        println(solve(inputs))
    }
}

private fun solve(inputs: List<Int>): String {
    val sorted = inputs.sorted()

    return when {
        (sorted[0] + sorted[1] > sorted[2]) -> {
            when {
                (sorted[0] == sorted[2]) -> "Equilateral"
                (sorted[0] == sorted[1] || sorted[1] == sorted[2]) -> "Isosceles"
                else -> "Scalene"
            }
        }
        else -> "Invalid"
    }
}